File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/events/AccountNotificationReceivedEvent.java
package com.hedvig.paymentservice.domain.trustlyOrder.events;

import lombok.Value;

@Value
public class AccountNotificationReceivedEvent {

    String notificationId;
    String trustlyOrderId;

    String accountId;
    String address;
    String bank;
    String city;
    String clearingHouse;
    String descriptor;
    Boolean directDebitMandate;
    String lastDigits;
    String name;
    String personId;
    String zipCode;


}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/events/AccountNotificationReceivedEvent.kt
package com.hedvig.paymentservice.domain.trustlyOrder.events

import java.util.UUID
import lombok.Value

@Value
class AccountNotificationReceivedEvent {

    val hedvigOrderId: UUID? = null
    val memberId: String? = null

    val notificationId: String? = null
    val trustlyOrderId: String? = null

    val accountId: String? = null
    val address: String? = null
    val bank: String? = null
    val city: String? = null
    val clearingHouse: String? = null
    val descriptor: String? = null
    val directDebitMandate: Boolean? = null
    val lastDigits: String? = null
    val name: String? = null
    val personId: String? = null
    val zipCode: String? = null
}
